pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -111,6 +111,130 @@ func TestMetricsQueryRequest(t *testing.T) {
 	}
 }
 
+func TestMetricsQuery_MinMaxTime(t *testing.T) {
+
+	rangeVectorDurationStr := "5m"
+	rangeVectorDuration, _ := time.ParseDuration(rangeVectorDurationStr)
+	rangeVectorDurationMS := rangeVectorDuration.Milliseconds()
+
+	offsetDurationStr := "1h"
+	offsetDuration, _ := time.ParseDuration(offsetDurationStr)
+	offsetDurationMS := offsetDuration.Milliseconds()
+
+	for _, testCase := range []struct {
+		name         string
+		metricsQuery MetricsQueryRequest
+		expectedMinT int64
+		expectedMaxT int64
+		expectedErr  error
+	}{
+		{
+			name: "range query: without range vector, without offset",
```

Comment:
after adding the test cases for @ modifier I feel like we are hitting diminishing returns on the test cases, since the only thing we are doing to calculate minT and maxT is calling Prometheus builtins that I think we can assume are more than well-tested enough.

more important I think and still TODO is to test that the transform funcs `WithQuery` and `WithStartEnd` will result in different minT and maxT where appropriate.